src/VisualStudio/Core/Def/Implementation/LanguageClient/AbstractLanguageServerClient.cs

```diff
@@ -20,7 +20,6 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService
     internal abstract class AbstractLanguageServerClient : ILanguageClient
     {
         private readonly string? _diagnosticsClientName;
-        private readonly IDiagnosticService _diagnosticService;
```

Comment:
all the lsp push diagnostic code is removed with this PR.

Reply:
The client is now sending a capability that tells us whether they support pull diagnostics (https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient/pullrequest/279454?_a=files) so if that is false, we should keep this code running.